saq/queue.py

```diff
@@ -384,7 +385,8 @@ async def dequeue(self, timeout=0):
                 "BLMOVE", self._queued, self._active, "RIGHT", "LEFT", timeout
             )
         if job_id is not None:
-            return await self.job(job_id)
+            job_key = Job.key_from_id(job_id.decode())
```

Comment:
not a fan of getting job_key from `job_id` here with the `decode` and the `split` in `Job.key_from_id`

Reply:
let's not do this, we should just have a second function or call job with directly 

job(id=id) which bypasses the logic

def job(key=None, id=None) something like this?

Reply:
How about a private `_get_job_by_id` (or something like that) method that expects `id`.
The public `job` method can just wrap that method, and this method can call it directly.